Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -304,18 +306,40 @@ def _validate_tests_in_id_set(self, tests: Iterable[str]):
             not_found_string = ', '.join(sorted(not_found))
             logger.warning(f'{len(not_found)} tests were not found in id-set: \n{not_found_string}')
 
-    @staticmethod
-    def _validate_support_level_is_xsoar(content_item: ContentItem) -> None:
-        if support_level := PACK_MANAGER.get_support_level(content_item.pack_id) != 'xsoar':
-            raise NonXsoarSupportedPackException(content_item.pack_id, support_level)
+    def validate_content_item_compatibility(self, content_item: ContentItem) -> None:
+        self.__validate_support_level_is_xsoar(content_item.pack_id, content_item.version_range)
+        self.__validate_marketplace_compatibility(content_item.marketplaces, content_item.path)
+
+    def validate_id_set_item_compatibility(self, id_set_item: IdSetItem) -> None:
+        # id_set_item objects may not have pack_id or path
+        if not (pack_id := id_set_item.pack_id or find_pack_folder(id_set_item.path).name):
+            raise RuntimeError(f'could not find pack of {id_set_item.name}')
+        self.__validate_support_level_is_xsoar(pack_id, id_set_item.version_range)
+        self.__validate_marketplace_compatibility(id_set_item.marketplaces, id_set_item.path)
 
-    def _collect_pack(self, pack_id: str, reason: CollectionReason, reason_description: str,
-                      content_item_range: Optional[VersionRange] = None) -> CollectionResult:
-        pack = PACK_MANAGER[pack_id]
+    def _collect_pack(
+            self,
+            pack_id: str,
+            reason: CollectionReason,
+            reason_description: str,
+            content_item_range: Optional[VersionRange] = None,
+            allow_incompatible_marketplace: bool = False,
+    ) -> Optional[CollectionResult]:
+        pack_metadata = PACK_MANAGER.get_pack_metadata(pack_id)
+
+        try:
+            self.validate_content_item_compatibility(pack_metadata)
+        except NonXsoarSupportedPackException:
+            pass  # we do want to install packs in this case (tests are not collected in this case anyway)
```

Comment:
Doesnt this justify a message to the logs?

Reply:
yup